tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,945 @@
+            deployment_yaml = join(self.backup_dir, deployment + ".yaml")
+            templating.dump_data_to_temp_yaml(deployment_get, deployment_yaml)
+
+    def deployments_to_revert(self):
+        """
+        Gets mon, osd and mgr deployments to revert
+
+        Returns:
+            list: list of deployment paths to be reverted
+
+        """
+        to_revert_patches = (
+            get_deployments_having_label(
+                label=constants.OSD_APP_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            + get_deployments_having_label(
+                label=constants.MON_APP_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+            + get_deployments_having_label(
+                label=constants.MGR_APP_LABEL,
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )
+        )
+        logger.info(to_revert_patches)
+        to_revert_patches_path = []
+        for dep in to_revert_patches:
+            to_revert_patches_path.append(
+                join(self.backup_dir, dep["metadata"]["name"] + ".yaml")
+            )
+        return to_revert_patches_path
+
+    def get_ceph_keyrings(self):
+        """
+        Gets all ceph and csi related keyring from OCS secrets
+
+        """
+        mon_k = get_ceph_caps(["rook-ceph-mons-keyring"])
+        if config.ENV_DATA["platform"] == constants.ON_PREM_PLATFORMS:
+            rgw_k = get_ceph_caps(
+                ["rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-keyring"]
+            )
+        else:
+            rgw_k = None
+        mgr_k = get_ceph_caps(["rook-ceph-mgr-a-keyring"])
+        mds_k = get_ceph_caps(
+            [
+                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-keyring",
+                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-keyring",
+            ]
+        )
+        crash_k = get_ceph_caps(["rook-ceph-crash-collector-keyring"])
+        fs_node_k = get_ceph_caps([constants.CEPHFS_NODE_SECRET])
+        rbd_node_k = get_ceph_caps([constants.RBD_NODE_SECRET])
+        fs_provisinor_k = get_ceph_caps([constants.CEPHFS_PROVISIONER_SECRET])
+        rbd_provisinor_k = get_ceph_caps([constants.RBD_PROVISIONER_SECRET])
+
+        keyring_caps = {
+            "mons": mon_k,
+            "rgws": rgw_k,
+            "mgrs": mgr_k,
+            "mdss": mds_k,
+            "crash": crash_k,
+            "fs_node": fs_node_k,
+            "rbd_node": rbd_node_k,
+            "fs_provisinor": fs_provisinor_k,
+            "rbd_provisinor": rbd_provisinor_k,
+        }
+
+        for secret, caps in keyring_caps.items():
+            if caps:
+                with open(f"{self.keyring_dir}/{secret}.keyring", "w") as fd:
+                    fd.write(caps)
+                    self.keyring_files.append(f"{self.keyring_dir}/{secret}.keyring")
+
+    def mds_deployments_to_revert(self):
```

Comment:
should we combine all deployments_to_revert to a single function?

Reply:
Agree, made changes.